hwdb.d/70-dj-and-production.hwdb

```diff
@@ -0,0 +1,133 @@
+# This file is part of systemd.
+#
+# Database for DJ and music production controllers that should be accessible to the seat owner.
+#
+# To add local entries, copy this file to
+#   /etc/udev/hwdb.d/
+# and add your rules there. To load the new rules execute (as root):
+#   systemd-hwdb update
+#   udevadm trigger
+
+################
+# Ableton
+################
+
+# Push 2
+usb:v2982p1967*
+ ID_DJ_AND_PRODUCTION=1
+
+################
+# Eks
+################
+
+# Otus
+usb:v1157p300*
+ ID_DJ_AND_PRODUCTION=1
+
+#############################
+# Hercules (Guillemot Corp)
+#############################
+
+# DJ Console MP3e2
+usb:v06f8pb105*
+ ID_DJ_AND_PRODUCTION=1
+
+# DJ Console MP3 LE / Glow
+usb:v06f8pb120*
+ ID_DJ_AND_PRODUCTION=1
+
+# DJ Console Mk2
+usb:v06f8pb100*
+ ID_DJ_AND_PRODUCTION=1
+
+# DJ Console Mk4
+usb:v06f8pb107*
+ ID_DJ_AND_PRODUCTION=1
+
+#####################
+# Native Instruments
+#####################
+
+# Maschine 2
+usb:v17ccp1140*
+ ID_DJ_AND_PRODUCTION=1
+
+# Maschine 2 Mikro
+usb:v17ccp1110*
+ ID_DJ_AND_PRODUCTION=1
+
+# Maschine 2 Studio
+usb:v17ccp1300*
+ ID_DJ_AND_PRODUCTION=1
+
+# Maschine Jam
+usb:v17ccp1500*
+ ID_DJ_AND_PRODUCTION=1
+
+# Maschine 3
+usb:v17ccp1600*
+ ID_DJ_AND_PRODUCTION=1
+
+# Traktor Kontrol D2
+usb:v17ccp1400*
+ ID_DJ_AND_PRODUCTION=1
+
+# Traktor Kontrol F1
+usb:v17ccp1120*
+ ID_DJ_AND_PRODUCTION=1
+
+# Traktor Kontrol S2 Mk2
+usb:v17ccp1320*
+ ID_DJ_AND_PRODUCTION=1
+
+# Traktor Kontrol S2 Mk3
+usb:v17ccp1710*
+ ID_DJ_AND_PRODUCTION=1
+
+# Traktor Kontrol S3
+usb:v17ccp1900*
+ ID_DJ_AND_PRODUCTION=1
+
+# Traktor Kontrol S4 Mk2
+usb:v17ccp1310*
+ ID_DJ_AND_PRODUCTION=1
+
+# Traktor Kontrol S4 Mk3
+usb:v17ccp1720*
+ ID_DJ_AND_PRODUCTION=1
+
+# Traktor Kontrol S5
+usb:v17ccp1420*
+ ID_DJ_AND_PRODUCTION=1
+
+# Traktor Kontrol S8
+usb:v17ccp1370*
+ ID_DJ_AND_PRODUCTION=1
+
+# Traktor Kontrol X1 Mk2
+usb:v17ccp1220*
+ ID_DJ_AND_PRODUCTION=1
+
+# Traktor Kontrol Z1
+usb:v17ccp1210*
+ ID_DJ_AND_PRODUCTION=1
+
+# Traktor Kontrol Z2
+usb:v17ccp1130*
+ ID_DJ_AND_PRODUCTION=1
+
+###############
+# Pioneer
+###############
+
+# CDJ 850
+usb:v834p157*
+ ID_DJ_AND_PRODUCTION=1
+
+# CDJ 900
+usb:v8e4p14b*
+ ID_DJ_AND_PRODUCTION=1
+
+# CDJ 2000 (not NXS)
+usb:v834p14a*
+ ID_DJ_AND_PRODUCTION=1
```

Comment:
@Swiftb0y do you have the USB vendor and product ID for the CDJ 2000 NXS 2?

Reply:
I don't think so unfortunately. I just added the new VendorID for alphatheta devices (the pioneer DJ rebrand) to our udev rules file. https://github.com/mixxxdj/mixxx/pull/2609

Reply:
Nevermind, I found an old pcap. It seems the device is posing as multiple devices. From what I could tell, the device has an id of `0x005` for the HID endpoint. it seems to have multiple different endpoints though. For example it seems to contain a Hi-speed Hub with a productID of `0x0006` and another vendor specific device/endpoint with `0x0007`

Reply:
I'm not interested in the endpoints for this, only the vendor and product ID. The S4 Mk3 also presents itself as two separate devices with one of them being a USB hub.

Reply:
I only have a pcap of the device descriptor query process. Feel free to take a look yourself: 
[Pioneer-CDJ-2000NXS2-Devicedescriptors.zip](https://github.com/systemd/systemd/files/7071216/Pioneer-CDJ-2000NXS2-Devicedescriptors.zip)

Reply:
Do you have any records of the `lsusb` output?

Reply:
Nevermind I found it in that Wireshark capture.

Reply:
unfortunately not. I know that would've been the easiest way to read the product id, but unfortunately I don't have that.

Reply:
I found a different set of captures that contains the entire conversation when the device is plugged in. From that I could read some more info. As I already said  `0x0006` is the USB hub, while `0x0005` is the device which contains the endpoints for audio and HID. `0x0007` seems to be a USB ethernet device. I'm not sure which of these devices are required for the CDJ to function properly, but I guess you can just add all three productIDs to the whitelist?

Reply:
Only the device with the HID endpoint PID `0x0005` needs to be listed here.